core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -430,6 +431,7 @@ public PlanOptimizers(
                                 .addAll(columnPruningRules)
                                 .addAll(projectionPushdownRules)
                                 .addAll(ImmutableSet.of(
+                                        new PreferWritePartitioning(),
```

Comment:
To be honest I have no idea where to put it here, so it landed at the beginning.

Reply:
@martint suggestions?

Reply:
Move it before or after reorder joins is applied. The assumption is that for reorder joins we should have the most accurate stats:
* `PredicatePushDown` has been run with `useTableProperties`
* later in the planning exchange nodes are added which might confuse stats.

Please add a comment why there.